### PR TITLE
Profiler callback

### PR DESCRIPTION
**Pull request by ditwoo**
## Before submitting (checklist)

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [x] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [x] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [x] Did you check the docs with `make check-docs`?
- [x] Did you write any new necessary tests?
- [x] Did you check that your code passes the unit tests `pytest .` ? 
- [x] Did you add your new functionality to the docs?
- [x] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?
- [ ] Did you run [colab minimal CI/CD](https://colab.research.google.com/drive/1JCGTVvWlrIsLXMPRRRSWiAstSLic4nbA) with `latest` and `minimal` requirements?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### FAQ
Please review the FAQ before submitting an issue:
- [ ] I have read the [documentation and FAQ](https://catalyst-team.github.io/catalyst/)
- [ ] I have reviewed the [minimal examples section](https://github.com/catalyst-team/catalyst#minimal-examples)
- [ ] I have checked the [changelog](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md) for main framework updates
- [ ] I have read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)
- [ ] I have joined [Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion

### CODE REVIEW

File: catalyst/callbacks/profiler.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            table_txt = self.stats.table(sort_by="cpu_time_total", row_limit=100)
+
+            with TemporaryDirectory() as tmp_dir:
+                artifact_path = os.path.join(tmp_dir, "profiler_table.txt")
+                with open(artifact_path, "w") as f:
+                    f.write(table_txt)
+                for logger in loggers_:
+                    logger.log_artifact(
+                        tag="profiler", artifact="profiler.txt", path_to_artifact=artifact_path,
+                    )
```
Comment (Scitator) [inline catalyst/callbacks/profiler.py]: maybe, let's use `runner.log_artifact` instead?

Comment (dmdr-sg) [reply catalyst/callbacks/profiler.py:178]: Yeah, I'll change that.
Btw, will it work correctly when the user somehow doesn't specified loggers (i.e. dict with loggers is empty)?

Comment (ditwoo) [reply catalyst/callbacks/profiler.py]: Yeah, I'll change that.
Btw, will it work correctly when dictionary with loggers is empty?

Comment (Scitator) [reply catalyst/callbacks/profiler.py]: yup :)

**Pull request merged**